### PR TITLE
PYIC-7641: journey routing when user indicates DL details are incorrect and drops out of DL CRI during auth source check

### PR DESCRIPTION
**Pull request by thebauSoftwire**
### What changed
- routing the user to new screens for DL auth source check
- clearing DCMAW VC after dropping out of DL CRI during DL auth source check
- update unit tests + api tests

### Why did it change
If a user indicates that their DL details are incorrect (e.g. bc of bad scan) then they should be routed to new DL auth source check pages which gives them options to try proving their identity in other ways or return to the RP.
The DCMAW VC gained during their session is also deleted to avoid issues with data integrity further down the journey/

### Issue tracking

- [PYIC-7641](https://govukverify.atlassian.net/browse/PYIC-7641)

[PYIC-7641]: https://govukverify.atlassian.net/browse/PYIC-7641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature
Comment (Joe-Edwards-GDS) [inline api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature]: (We don't actually care about the DCMAW VC if they return with a P0 - they don't have an identity either way, but it doesn't hurt to check!)

File: api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature
Comment (Joe-Edwards-GDS) [inline api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature]: I've mentioned to Chris, but I'm campaigning to combine these routes together (by performing the same COI check), which should halve a lot of these test cases 🤞

File: lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/nested-journeys/app-doc-check.yaml
Comment (Joe-Edwards-GDS) [inline lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/nested-journeys/app-doc-check.yaml:48]: Do you think the other reset should be DCMAW as well? I don't think there are currently any scenarios where it matters, but it feels more consistent.

(The D02 only really invalidates the DCMAW VC)

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Looks good to me - just a couple of minor comments in the routing that might be worth tweaking.

File: api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature
Comment (thebauSoftwire) [reply api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature]: oooo nice! Should I get rid of these tests then or leave in for the meantime?

Comment (Joe-Edwards-GDS) [reply api-tests/features/dl-auth-source-checks/dl-cri-dropout.feature]: Whichever is easiest tbh - if you keep it in then you could make a note on https://govukverify.atlassian.net/browse/PYIC-7701, or just squash them pre-emptively.

**Review by Joe-Edwards-GDS — APPROVED** [review]
Thanks! Maybe wait until after lunch to merge, so we avoid the no-photo-id run-through

**Pull request merged**
